src/providers/sql.js

```diff
@@ -0,0 +1,90 @@
+var d3 = require("d3");
+
+function SQLProvider(options) {
+	this.sql_api_template = options.sql_api_template || 'http://{user}.cartodb.com';
+	this.user = options.user;
+	this.table = options.table;
+	this.format = options.format;
+	this.tileCache = {};
+}
+
+SQLProvider.prototype = {
+	getTile: function(tilePoint, callback){
+    var tileData = this.tileCache[tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y];
+    if (tileData) {
+      callback(tilePoint, tileData);
+    }
+    else{
+      function tile2lon(x,z) {
```

Comment:
I know you are just moving code but this functions are pure, can we extract them to a geo package?